### PR TITLE
Automated cherry pick of #120090: Handle edge cases in seat demand stats

### PR DESCRIPTION
**Pull request by Swizzmaster**
kind/bug

Cherry pick of #120090 on release-1.26.

#120090: Handle edge cases in seat demand stats

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
NONE
```

### CODE REVIEW

**Comment by alexzielenski** [comment]
/cc @MikeSpreitzer @wojtek-t 
could you take a look if you have time?
/triage accepted

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm
/approve
Thanks!

**Comment by xmudrii** [comment]
@MikeSpreitzer We also need #122513 and #122514 lgtm-ed and approved before we can cherry-pick-approve this PR

**Review by Verolop — COMMENTED** [review]
/kind bug

**Pull request merged**
